packages/block-editor/src/components/writing-flow/index.js

```diff
@@ -209,13 +212,53 @@ export default function WritingFlow( { children } ) {
 		hasMultiSelection,
 		blocks,
 		isNavigationMode,
-	} = useSelect( selector );
+		isSelectionEnabled,
+		blockSelectionStart,
+	} = useSelect( selector, [] );
 	const {
 		multiSelect,
 		selectBlock,
 		clearSelectedBlock,
+		setNavigationMode,
 	} = useDispatch( 'core/block-editor' );
 
+	function onMouseDown( event ) {
```

Comment:
The logic below belongs here in my opinion, as this component handles multi selection as well as navigation mode already. Perhaps we can split this is the future, but a lot of it seems tied together.